File: app/src/main/java/io/github/alexeychurchill/stickynotes/activity/main/friends/fragment/UserFriendRequestListFragment.java
package io.github.alexeychurchill.stickynotes.activity.main.friends.fragment;

import android.support.v4.app.Fragment;

import java.util.List;

import io.github.alexeychurchill.stickynotes.model.FriendRequest;
import io.github.alexeychurchill.stickynotes.model.ServiceResponse;
import retrofit2.Call;

/**
 * User friend request list fragment
 */

public class UserFriendRequestListFragment extends FriendRequestListFragment {
    @Override
    protected Call<ServiceResponse<List<FriendRequest>>> getLoadDataPageCall(int page) {
        return null;
    }
}

File: app/src/main/java/io/github/alexeychurchill/stickynotes/contacts/UserFriendRequestListFragment.kt
package io.github.alexeychurchill.stickynotes.contacts

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.compose.ui.platform.ComposeView
import androidx.fragment.app.Fragment
import androidx.fragment.app.viewModels
import dagger.hilt.android.AndroidEntryPoint
import io.github.alexeychurchill.stickynotes.contacts.presentation.ContactRequestsViewModel
import io.github.alexeychurchill.stickynotes.contacts.ui.OutcomingRequestScreen
import io.github.alexeychurchill.stickynotes.core.ui.StickyNotesTheme

/**
 * User friend request list fragment
 */
@AndroidEntryPoint
class UserFriendRequestListFragment : Fragment() {

    private val viewModel by viewModels<ContactRequestsViewModel>()

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View = ComposeView(requireContext())

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        val composeView = (view as? ComposeView) ?: return
        composeView.setContent {
            StickyNotesTheme {
                OutcomingRequestScreen(viewModel)
            }
        }
    }
}
